### ISSUE TITLE
New top of library card not revealed until old card resolves; should be revealed as soon as old card is put on stack

### ISSUE THREAD

**Issue by Zerrisx** [magefree/mage#6735]
If you have an effect (Future Sight, Experimental Frenzy, Conspicuous Snoop) that allows you to cast cards from the top of your library, the new card should be visible as soon as the old one is paid for and put on the stack.

Currently, that's not true:
![image](https://user-images.githubusercontent.com/59861365/85915657-feac9e80-b80e-11ea-8544-3c5c94c2e890.png)

**Comment by jeffwadsworth** [comment]
Ruling:  If the top card of your library changes while you’re casting a spell, playing a land, or activating an ability (most likely because that top card is the card you’re playing), the new top card won’t be revealed until you finish doing so.

**Comment by Zerrisx** [comment]
Correct, it should technically be facedown from when you declare an intention to cast the old card to when it is paid for and goes on the stack.
